test/protocol_test.py

```diff
@@ -448,9 +448,6 @@ def test_time_constraint(self, dummy_protocol):
         p.cover(plate_2)
         time_point_2 = p.get_instruction_index()
 
-        with pytest.raises(AttributeError):
-            p.time_constraints  # pylint: disable=pointless-statement
-
```

Comment:
fyi: I believe the main intent here was to ensure we don't unnecessarily serialize the field if its empty.

Something you could do is to replicate a check for that by examining the `p.as_dict` output, and ensuring the `time_constraints` key is not present.